packages/picasso.js/src/core/data/dataset.js

```diff
@@ -217,49 +217,155 @@ ds.util = {
+ * @property {DataExtraction~SortCallback} [sort] Callback function to sort the extracted data items
+ * @example
+{
+  extract: [{
+    source: 'Products',
+    field: 'Product',
+    value: d => d.name,
+    label: d => `<${d.name}>`
+    props: {
+      year: { field: 'Year' }
+      num: { field: 'Sales' }
+    }
+  }],
+  filter: d => d.label !== 'Sneakers', // extract everything except Sneakers
+  sort: (a, b) => a.label > b.label ? -1 : 1, // sort descending
+}
+ */
+
+/**
+ * @typedef {object} DataFieldExtraction
+ * @property {string} source - Which data source to extract from
+ * @property {string} field - The field to extract data from
+ * @property {DataExtraction~Extract~ValueFn} [value] - The field value accessor
+ * @property {DataExtraction~Extract~LabelFn} [label] - The field label accessor
+ * @example
+ * {
+ *  source: 'Products',
+ *  field: 'Sales',
+ *  value: (val) => Math.round(val),
+ *  label: (val) => `<${val}>`
+ * }
+ */
+
+/**
+ * Data extraction definition. Define how and what kind of data should be extracted from a `DataSource`.
+ * @typedef {object} DataExtraction~Extract
+ * @property {string} source - Which data source to extract from
+ * @property {string} field - The field to extract data from
+ * @property {DataExtraction~Extract~ValueFn} [value] - The field value accessor
+ * @property {DataExtraction~Extract~LabelFn} [label] - The field label accessor
+ * @property {DataExtraction~Extract~TrackByFn} [trackBy] - Track by value accessor
+ * @property {DataExtraction~Extract~ReduceFn} [reduce] - Reducer function
+ * @property {DataExtraction~Extract~ReduceLabelFn} [reduceLabel] - Label reducer function
+ * @property {DataExtraction~Extract~FilterFn} [filter] - Filter function
+ * @property {object.<string, DataExtraction~Extract~Props>} [props] - Additional properties to add to the extracted item
+ * @example
+ * {
+    source: 'Products',
+    field: 'Product',
+    value: (val) => val,
+    label: (val) => `<${val}>`
+    props: {
+      year: { field: 'Year' }
+      num: { field: 'Sales' }
+    }
+  }
+ */
+
+/**
+ * @typedef {object} DataExtraction~Extract~Props
  * @property {string} field - The field to extract data from
- * @property {DataExtractConfig~ValueFn} [value] - The field value accessor
- * @property {DataExtractConfig~LabelFn} [label] - The field label accessor
- * @property {DataExtractConfig~TrackByFn} [trackBy] - Track by value accessor
- * @property {DataExtractConfig~ReduceFn} [reduce] - Reducer function
- * @property {DataExtractConfig~ReduceLabelFn} [reduceLabel] - Label reducer function
- * @property {DataExtractConfig~FilterFn} [filter] - Filter function
- * @property {object} [props] - Additional properties to add to the extracted item
+ * @property {DataExtraction~Extract~ValueFn} [value] - The field value accessor
+ * @property {DataExtraction~Extract~LabelFn} [label] - The field label accessor
+ * @example
```

Comment:
why is reduce and reduceLabel removed here as they are suppored on this level

Reply:
I'll double check if you can actually use reduce on this level as well

Reply:
Doesn't look like it

Reply:
there is a lot of places in la-vie that set reduceLabel on this level to avoid generating unused labels (for performace)
could it be that is it only relevant for stacked data (using picasso-plugin-q)?

Reply:
What level are you referring to? You comment is on the `props` property, is that the level your are referring to?

Reply:
on a property inside porps

ex:
```
props: {
  year: {
    field: 'Year',
    reduceLabel: 'none',
  },
}
```

it also reminds me that reduceLabel can be 
a function, a string refering to a common predefined function or any other string (in this case 'none') to not have any reduceLabel function (in case there are no reduceLabel it will have a default one that do client side formatting)